actions/invite_actions.jsx

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import * as TeamActions from 'mattermost-redux/actions/teams';
+import {getTeamMember} from 'mattermost-redux/selectors/entities/teams';
+import {getChannelMembersInChannels} from 'mattermost-redux/selectors/entities/channels';
+import {joinChannel} from 'mattermost-redux/actions/channels';
+
+import {addUsersToTeam} from 'actions/team_actions';
+
+import {isGuest, localizeMessage} from 'utils/utils';
+
+export function sendMembersInvites(teamId, users, emails) {
+    return async (dispatch, getState) => {
+        if (users.length > 0) {
+            await dispatch(TeamActions.getTeamMembersByIds(teamId, users.map((u) => u.id)));
+        }
+        const state = getState();
+        const sent = [];
+        const notSent = [];
+        const usersToAdd = [];
+        for (const user of users) {
+            const member = getTeamMember(state, teamId, user.id);
+            if (member) {
+                notSent.push({user, reason: localizeMessage('invite.members.already-member', 'This person is already a team member.')});
+            } else if (isGuest(user)) {
+                notSent.push({user, reason: localizeMessage('invite.members.user-is-guest', 'Contact your admin to make this guest a full member.')});
+            } else {
+                usersToAdd.push(user);
+            }
+        }
+        if (usersToAdd.length > 0) {
+            let response;
+            try {
+                response = await dispatch(addUsersToTeam(teamId, usersToAdd.map((u) => u.id)));
+            } catch (e) {
+                response.error = localizeMessage('invite.members.unable-to-add-the-user-to-the-team', 'Unable to add the user to the team');
+            }
+            for (const user of usersToAdd) {
+                if (response.error) {
+                    notSent.push({user, reason: response.error});
+                } else {
+                    sent.push({user, reason: localizeMessage('invite.members.added-to-team', 'This member has been added to the team')});
+                }
+            }
+        }
+        if (emails.length > 0) {
+            let response;
+            try {
+                response = await dispatch(TeamActions.sendEmailInvitesToTeam(teamId, emails));
```

Comment:
Could it be possible to bypass the guest restriction simply adding them by email instead of searching for the user?

Reply:
You can send invitations to any email for guest or for members, but anyway when you going to use the invitations the restrictions are applied anyway, so... is not a big deal. Beside, probably, we going to implement the restriction in the "sendEmailInvites" server endpoint to return this error as soon as possible. But that will be done as another PR.